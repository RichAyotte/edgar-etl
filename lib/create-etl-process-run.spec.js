/**
 * @overview    Create ETL process run test
 * @author      Richard Ayotte
 * @copyright   Copyright Â© 2019 Richard Ayotte
 * @license     MIT License
 */

'use strict'

const connectionPool = require('./get-db-connection-pool')
const createEtlProcessRun = require('./create-etl-process-run')
const populateTestDatabase = require('./populate-test-database')
const resetTestDatabase = require('./reset-test-database')

beforeAll(async () => {
	await resetTestDatabase()
	await populateTestDatabase()
})

afterAll(async () => {
	await connectionPool.end()
})

describe('Create ETL Process run', () => {
	it(`Throws on invalid process id`, async () => {
		await expect(createEtlProcessRun({
			etlProcessId: 2
			, status: 'started'
		}))
		.rejects
		.toThrow(Error)
	})

	it(`Creates a process run`, async () => {
		await expect(createEtlProcessRun({
			etlProcessId: 1
			, status: 'started'
		}))
		.resolves
		.toMatchSnapshot()
	})
})
