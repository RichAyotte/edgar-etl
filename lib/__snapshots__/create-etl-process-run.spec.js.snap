// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create ETL Process run Creates a process run 1`] = `2`;
